books/book.js

```diff
@@ -30,10 +30,75 @@ function displayBook(book) {
   const buttonBuy = `<button class= 'buy'> Buy Now </button>`;
   const buttonBasket = `<button class = 'basket' > Add to basket </button>`;
   const buttonWish = `<button class = 'wishlist' > Add to wishlist </button>`;
+  const loginMsg = `<a class = 'login-msg' href = '../login/login.html'> Please Login To Buy Books </a>`;
   const bookInfoDiv = `<div class = 'book-info'> ${title} ${author} ${bookYear} ${topic} ${synopsis} </div>`;
   const bookCoverDiv = `<div class = 'book-cover-div'><img class = 'book-cover' src = ${book.cover_url}></div>`;
-  const bookPurchase = `<div class = 'book-purchase'> ${bookPrice} ${bookPurchaseH3} ${bookPurchaseP} ${buttonBuy} ${buttonBasket} ${buttonWish}</div>`;
+  const bookPurchase = `<div class = 'book-purchase'> ${bookPrice} ${bookPurchaseH3} ${bookPurchaseP} ${buttonBuy} ${buttonBasket} ${buttonWish} ${loginMsg}  </div>`;
   currentBook = `<div class = 'book'>${bookInfoDiv} ${bookCoverDiv}  ${bookPurchase}</div>`;
 
   main.innerHTML = currentBook;
+  let msg = document.querySelector(".login-msg");
+  let buttonsBuy = document.querySelectorAll(".book button");
+  let usernameSessionValue = sessionStorage.getItem("username");
+  let loginBtn = document.querySelector(".login-btn");
+  let username = document.querySelector(".username");
+  let signOutBtn = document.querySelector(".sign-out");
+  if (usernameSessionValue) {
+    buttonsBuy.forEach((btn) => {
+      btn.style.display = "block";
+      username.style.display = "block";
+      username.innerHTML = usernameSessionValue + '<i class="fas fa-user"></i>';
+      loginBtn.style.display = "none";
+      signOutBtn.style.display = "block";
+    });
+    msg.style.display = "none";
+  } else {
+    buttonsBuy.forEach((btn) => {
+      btn.style.display = "none";
+    });
+  }
+
+  signOutBtn.addEventListener("click", signOut);
+  function signOut() {
+    window.open("../index.html", "_self");
+    sessionStorage.removeItem("username");
+    sessionStorage.removeItem("password");
```

Comment:
I see this code in one form or in another in every js file. We generally don't want to repeat ourselves as programmers. I would suggest creating a navbar.js file in which you would define this behaviour and then just import that script to every html file

Reply:
Alright. I will try to fix tomorrow